src/fileio.cpp

```diff
@@ -977,55 +973,67 @@ bool DoScanWorkingDirectory()
 	return _searchpaths[SP_PERSONAL_DIR] != tmp;
 }
 
+/**
+ * Gets the home directory of the user.
+ * May return an empty string in the unlikely scenario that the home directory cannot be found.
+ * @return User's home directory
+ */
+static std::string GetHomeDir()
+{
+	std::string homedir;
+#ifdef __HAIKU__
+	BPath path;
+	find_directory(B_USER_SETTINGS_DIRECTORY, &path);
+	homedir = path.Path();
+#else
+	char *home_env = getenv("HOME");
+	if (home_env != nullptr) {
+		homedir = home_env;
+	} else  {
+		const struct passwd *pw = getpwuid(getuid());
+		if (pw != nullptr) {
+			homedir = pw->pw_dir;
+		}
+	}
+#endif
+	return homedir;
+}
+
 /**
  * Determine the base (personal dir and game data dir) paths
  * @param exe the path to the executable
  */
 void DetermineBasePaths(const char *exe)
 {
 	std::string tmp;
-#if defined(WITH_XDG_BASEDIR) && defined(WITH_PERSONAL_DIR)
-	const char *xdg_data_home = xdgDataHome(nullptr);
-	tmp = xdg_data_home;
-	tmp += PATHSEP;
-	tmp += PERSONAL_DIR[0] == '.' ? &PERSONAL_DIR[1] : PERSONAL_DIR;
-	free(xdg_data_home);
+#ifdef USE_XDG
+	std::string homedir = GetHomeDir();
+	const char *xdg_data_home = getenv("XDG_DATA_HOME");
+	if (xdg_data_home != nullptr) {
+		tmp = xdg_data_home;
+		tmp += PATHSEP;
+		tmp += PERSONAL_DIR[0] == '.' ? &PERSONAL_DIR[1] : PERSONAL_DIR;
+	} else if (!homedir.empty()) {
+		tmp = homedir;
+		tmp += PATHSEP ".local" PATHSEP "share" PATHSEP;
+		tmp += PERSONAL_DIR[0] == '.' ? &PERSONAL_DIR[1] : PERSONAL_DIR;
+	}
```

Comment:
`tmp` can be empty; is it intended to assign this to `SP_PERSONAL_DIR_XDG`?

Reply:
It starts off empty, so no great loss

Reply:
Especially because it is named `tmp`, aren't you afraid people will be confused? In other places, like a bit lower, you see that they do similar what you do, but do an explicit `clear()` on the `_searchpaths`. That does make the intention of that it is not by accident more clear.

But possibly, and this is not something for this PR, this function needs revising .. as `tmp` is just a silly variable name :P Moving it into smaller scopes would be so much better and less confusing :D